### PR TITLE
writev

### PR DESCRIPTION
**Pull request by Jan-Jan**

### CODE REVIEW

File: pino-eventhub.js [hunk 1]
```diff
@@ -67,19 +74,44 @@ function pinoEventHub (opts) {
...
-      if (body) {
-        var req = https.request(options, callback(done))
+    writev: function (lines, done) {
+      // https://docs.microsoft.com/en-us/rest/api/eventhub/send-batch-events
+      const events = lines
+        .map(line => {
+          // check if console output is a string or object
+          var parsed = new Parse(line)
+          if (parsed.err) {
+            return `{"Body":"${line}"}`
```
Comment (mcollina) [inline pino-eventhub.js]: It's better if you do a JSON.stringify here, otherwise you might get into escaping issues

File: pino-eventhub.js [hunk 2]
```diff
@@ -35,24 +36,30 @@ function pinoEventHub (opts) {
     return line
   })
 
-  var url = decodeURIComponent(opts.sr) + '/messages'
-  var options = {
+  const url = decodeURIComponent(opts.sr) + '/messages'
+  const options = {
```
Comment (mcollina) [inline pino-eventhub.js:40]: we should also have an http.Agent created, with a configurable number of sockets attached.

File: pino-eventhub.js [hunk 3]
```diff
@@ -67,19 +74,44 @@ function pinoEventHub (opts) {
...
+    writev: function (lines, done) {
+      // https://docs.microsoft.com/en-us/rest/api/eventhub/send-batch-events
+      const events = lines
+        .map(line => {
+          // check if console output is a string or object
+          var parsed = new Parse(line)
+          if (parsed.err) {
+            return `{"Body":"${line}"}`
+          }
+          return `{"Body":${line}}`
```
Comment (mcollina) [inline pino-eventhub.js]: is Body mandatory in the bulk API?

**Review by mcollina — COMMENTED** [review]
Good job, a couple of nits and questions

File: pino-eventhub.js [hunk 4]
```diff
@@ -35,24 +36,30 @@ function pinoEventHub (opts) {
...
 
-  var url = decodeURIComponent(opts.sr) + '/messages'
-  var options = {
+  const url = decodeURIComponent(opts.sr) + '/messages'
+  const options = {
     method: 'POST',
     host: opts.host.slice(8), // remove 'https://'
     port: opts.port,
-    path: '/' + opts.eh + '/messages',
+    path: '/' + opts.eh + '/messages?timeout=60&api-version=2014-01',
```
Comment (davidmarkclements) [inline pino-eventhub.js:44]: should the timeout be configurable?

File: pino-eventhub.js (hunk 3 above)
Comment (Jan-Jan) [reply pino-eventhub.js]: From how I understand their docs, yes it is.

File: pino-eventhub.js (hunk 2 above)
Comment (Jan-Jan) [reply pino-eventhub.js:40]: I see you already created an issue for http.Agent, thank you.

File: pino-eventhub.js (hunk 1 above)
Comment (Jan-Jan) [reply pino-eventhub.js]: I don't think I can use a JSON.stringify here -- it is already a string.
However, I see your point about malformed json being piped in...

File: pino-eventhub.js (hunk 4 above)
Comment (Jan-Jan) [reply pino-eventhub.js:44]: Sure, I could make that configurable.

File: pino-eventhub.js (hunk 1 above)
Comment (mcollina) [reply pino-eventhub.js]:
```
JSON.stringify({ Body: line })
```

if `line` includes any `"`, you will have broken JSON.

File: pino-eventhub.js (hunk 4 above)
Comment (Jan-Jan) [reply pino-eventhub.js:44]: I created an issue for this

**Pull request merged**